pkg/errors/build_errors.go

```diff
@@ -1013,6 +1013,205 @@ func NewGeneralUnknownError() Error {
+func NewSecretFailedValueFromStdin() Error {
+	return NewSecretFailedValueFromStdinBuilder().Build()
+}
+
+// SecretMissingNameErrorCode is the code for an instance of "missing_name_error".
+const SecretMissingNameErrorCode = "rcli_secret_missing_name_error"
+
+// IsSecretMissingNameError tests whether a given error is an instance of "missing_name_error".
+func IsSecretMissingNameError(err errawr.Error) bool {
+	return err != nil && err.Is(SecretMissingNameErrorCode)
+}
+
+// IsSecretMissingNameError tests whether a given error is an instance of "missing_name_error".
+func (External) IsSecretMissingNameError(err errawr.Error) bool {
+	return IsSecretMissingNameError(err)
+}
+
+// SecretMissingNameErrorBuilder is a builder for "missing_name_error" errors.
+type SecretMissingNameErrorBuilder struct {
+	arguments impl.ErrorArguments
+}
+
+// Build creates the error for the code "missing_name_error" from this builder.
+func (b *SecretMissingNameErrorBuilder) Build() Error {
+	description := &impl.ErrorDescription{
+		Friendly:  "Please provide a workflow name.",
+		Technical: "Please provide a workflow name.",
+	}
+
+	return &impl.Error{
+		ErrorArguments:   b.arguments,
+		ErrorCode:        "missing_name_error",
+		ErrorDescription: description,
+		ErrorDomain:      Domain,
+		ErrorMetadata:    &impl.ErrorMetadata{},
+		ErrorSection:     SecretSection,
+		ErrorSensitivity: errawr.ErrorSensitivityNone,
+		ErrorTitle:       "Missing workflow name error",
+		Version:          1,
+	}
+}
+
+// NewSecretMissingNameErrorBuilder creates a new error builder for the code "missing_name_error".
+func NewSecretMissingNameErrorBuilder() *SecretMissingNameErrorBuilder {
+	return &SecretMissingNameErrorBuilder{arguments: impl.ErrorArguments{}}
+}
+
+// NewSecretMissingNameError creates a new error with the code "missing_name_error".
+func NewSecretMissingNameError() Error {
+	return NewSecretMissingNameErrorBuilder().Build()
+}
+
+// SecretNameReadErrorCode is the code for an instance of "name_read_error".
+const SecretNameReadErrorCode = "rcli_secret_name_read_error"
+
+// IsSecretNameReadError tests whether a given error is an instance of "name_read_error".
+func IsSecretNameReadError(err errawr.Error) bool {
+	return err != nil && err.Is(SecretNameReadErrorCode)
+}
+
+// IsSecretNameReadError tests whether a given error is an instance of "name_read_error".
+func (External) IsSecretNameReadError(err errawr.Error) bool {
+	return IsSecretNameReadError(err)
+}
+
+// SecretNameReadErrorBuilder is a builder for "name_read_error" errors.
+type SecretNameReadErrorBuilder struct {
+	arguments impl.ErrorArguments
+}
+
+// Build creates the error for the code "name_read_error" from this builder.
+func (b *SecretNameReadErrorBuilder) Build() Error {
+	description := &impl.ErrorDescription{
+		Friendly:  "Could not read secret name. Please supply a valid name.",
+		Technical: "Could not read secret name. Please supply a valid name.",
```

Comment:
I know this is what the underlying structure expects, but all of these messages are the same string for both "friendly" and "technical" - seems like a smell.

Reply:
I would say that in this case Errawr perhaps hasn't provided the best value. But when you define a secret, you can in fact for the description provide both. For example, instead of:

```yaml
description: You did a bad thing!
```

You can write:

```yaml
description:
  friendly: You did a bad thing!
  technical: A bad thing occurred at {{location}} because {{subsystem}} is unavailable.
```

It's up to the individual writing the error to determine when it's appropriate to provide this additional context as well as how to present it, though. High burden overall.